### PR TITLE
New option to prevent ~/.netrc modifications

### PR DESCRIPTION
**Pull request by avalentino**
* userneme and password can always be provided explicitly file
* none of the API functions attempts to modify the ~/.netrc file
* ~/.netrc setup mover to the main function of the CLI

### CODE REVIEW

File: eof/cli.py [hunk 1]
```diff
@@ -72,6 +96,11 @@ def cli(
     orbit_type: str,
     force_asf: bool,
     debug: bool,
+    asf_user: str = "",
+    asf_password: str = "",
+    cdse_user: str = "",
+    cdse_password: str = "",
+    save_credantials: bool = False,
```
Comment (scottstanie) [inline eof/cli.py:103]:
```suggestion
    save_credentials: bool = False,
```

File: eof/cli.py [hunk 2]
```diff
@@ -72,6 +96,11 @@ def cli(
     orbit_type: str,
     force_asf: bool,
     debug: bool,
+    asf_user: str = "",
+    asf_password: str = "",
+    cdse_user: str = "",
```
Comment (scottstanie) [inline eof/cli.py:101]:
```suggestion
    cdse_username: str = "",
```

Comment (scottstanie) [inline eof/cli.py:101]: actually these are `_user` here, but `_username` in `download.py`

File: eof/cli.py [hunk 3]
```diff
@@ -82,6 +111,12 @@ def cli(
     With no arguments, searches current directory for Sentinel 1 products
     """
     log._set_logger_handler(level=logging.DEBUG if debug else logging.INFO)
+    dryrun = not save_credantials
```
Comment (scottstanie) [inline eof/cli.py:114]:
```suggestion
    dryrun = not save_credentials
```

File: eof/cli.py [hunk 4]
```diff
@@ -63,6 +64,29 @@
...
+)
+@click.option(
+    "--asf-password",
+    help="ASF password. If not provided the program asks for it",
+)
+@click.option(
+    "--setup-netrc",
+    is_flag=True,
+    help="save credentials provided interactively in the ~/.netrc file if necessary",
+)
```
Comment (scottstanie) [inline eof/cli.py:89]: this one is missing from the function, but `setup_netrc` is also an import. maybe we can do an absolute import, or else name this something else

**Review by scottstanie — COMMENTED** [review]
i think it'll work with these changes, and you're probably right with the sentiment of this PR that it's better not to surprise people by modifying the .netrc without notifying

File: eof/cli.py (hunk 4 above)
Comment (avalentino) [reply eof/cli.py:89]: fixed

File: eof/cli.py (hunk 3 above)
Comment (avalentino) [reply eof/cli.py:114]: done

**Review by scottstanie — APPROVED** [review]

**Comment by scottstanie** [comment]
Oh should we also bump the version to 0.9.1?

**Comment by avalentino** [comment]
> Oh should we also bump the version to 0.9.1?

yes
Shall I do it in this PR?

**Pull request merged**